### PR TITLE
Placeholder REST controller to push analytics events.

### PR DESCRIPTION
**Pull request by afoucret**
Create the transport and REST action to be used to push analytics events.
Currently it is a no-op implementation that just return always OK.

@tutelaris will be able to plug the event emitter service here.

### CODE REVIEW

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/analytics/action/PostAnalyticsEventAction.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            this.collectionName = in.readString();
+        }
+
+        public String collectionName() {
+            return collectionName;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
```
Comment (kderusso) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/analytics/action/PostAnalyticsEventAction.java:52]: Should we validate that `collectionName` is non-empty, does not have any disallowed characters?

**Review by kderusso — APPROVED** [review]
Overall changes LGTM. I think it would be a good idea to add some yaml tests as well, that can be updated once the PR is more than a no-op. Thanks!

**Review by jimczi — COMMENTED** [review]
If this is needed for @tutelaris , is it possible to share this work with him directly? I don't like having boilerplates like this even if it's in a branch. Taking this commit into another branch should be straightforward, no?

**Comment by afoucret** [comment]
Replaced by https://github.com/elastic/elasticsearch/pull/94616

**Pull request closed without merging**